src/rust/ensogl/lib/text/src/component/area.rs

```diff
@@ -452,15 +682,99 @@ impl AreaData {
+            let min_pos_x  = self.lines.rc.borrow()[start_line].div_by_column(sel.start.column);
+            let max_pos_x  = self.lines.rc.borrow()[end_line].div_by_column(sel.end.column);
+            let logger     = Logger::sub(&self.logger,"cursor");
+            let min_pos_y  = -LINE_HEIGHT/2.0 - LINE_HEIGHT * start_line as f32;
+            let pos        = Vector2(min_pos_x,min_pos_y);
+            let width      = max_pos_x - min_pos_x;
+            let selection  = match selection_map.id_map.remove(&id) {
+                Some(selection) => {
+                    let select_left  = selection.width.simulator.target_value() < 0.0;
+                    let select_right = selection.width.simulator.target_value() > 0.0;
+                    let tgt_pos_x    = selection.position.simulator.target_value().x;
+                    let tgt_width    = selection.width.simulator.target_value();
+                    let mid_point    = tgt_pos_x + tgt_width / 2.0;
+                    let go_left      = pos.x < mid_point;
+                    let go_right     = pos.x > mid_point;
+                    let need_flip    = (select_left && go_left) || (select_right && go_right);
+                    if width == 0.0 && need_flip { selection.flip_sides() }
+                    selection.position.set_target_value(pos);
+                    selection
+                }
+                None => {
+                    let selection = Selection::new(&logger,&self.scene,do_edit);
+                    selection.shape.sprite.size.set(Vector2(4.0,20.0)); // FIXME hardcoded values
+                    selection.shape.letter_width.set(7.0); // FIXME hardcoded values
+                    self.add_child(&selection);
+                    selection.position.set_target_value(pos);
+                    selection.position.skip();
+                    let selection_network = &selection.network;
+                    let model = self.clone_ref(); // FIXME: memory leak. To be fixed with the below note.
+                    frp::extend! { selection_network
+                        // FIXME[WD]: This is ultra-slow. Redrawing all glyphs on each
+                        //            animation frame. Multiple times, once per cursor.
+                        eval_ selection.position.value (model.redraw());
+                    }
+                    selection
+                }
+            };
+
+            selection.width.set_target_value(width);
+            selection.edit_mode.set(do_edit);
+            selection.shape.start_time.set(time);
+            new_selection_map.id_map.insert(id,selection);
+            new_selection_map.location_map.entry(start_line).or_default().insert(sel.start.column,id);
+        }
+        *selection_map = new_selection_map;
+        }
+        self.redraw()
     }
 
-    pub fn line_count(&self) -> usize {
-        self.lines.len()
+    fn to_object_space(&self, screen_pos:Vector2) -> Vector2 {
```

Comment:
the term "object" means display object, right? It is not obvious at first glance. Perhaps docs could help.

Reply:
Yep, but this is pretty popular term in opengl https://stackoverflow.com/questions/16225815/object-world-camera-and-projection-spaces-in-opengl
In every book there is description of word space, eye space, object space. Anyway, I added docs :)